cmd/store/import.go

```diff
@@ -35,79 +36,157 @@ import (
-			clientConfig,
-			storeDataName,
-			storeData.Model,
-			format,
-		)
-		if err != nil {
-			return nil, err
-		}
+	clientConfig.StoreID = createStoreAndModelResponse.Store.Id
 
-		response = createStoreAndModelResponse
-		clientConfig.StoreID = createStoreAndModelResponse.Store.Id
-	} else {
-		store, err := fgaClient.GetStore(context.Background()).Execute()
-		if err != nil {
-			return nil, err //nolint:wrapcheck
-		}
+	return createStoreAndModelResponse, nil
+}
+
+func updateStore(
+	clientConfig fga.ClientConfig,
+	fgaClient client.SdkClient,
+	storeData *storetest.StoreData,
+	format authorizationmodel.ModelFormat,
+	storeID string,
+) (*CreateStoreAndModelResponse, error) {
+	store, err := fgaClient.GetStore(context.Background()).Execute()
+	if err != nil {
+		return nil, fmt.Errorf("failed to get store: %w", err)
+	}
 
-		response.Store = client.ClientCreateStoreResponse{
+	response := &CreateStoreAndModelResponse{
+		Store: client.ClientCreateStoreResponse{
 			CreatedAt: store.GetCreatedAt(),
 			Id:        store.GetId(),
 			Name:      store.GetName(),
 			UpdatedAt: store.GetUpdatedAt(),
-		}
+		},
+	}
+
+	authModel := authorizationmodel.AuthzModel{}
+	clientConfig.StoreID = storeID
+
+	err = authModel.ReadModelFromString(storeData.Model, format)
+	if err != nil {
+		return nil, fmt.Errorf("failed to read model: %w", err)
+	}
 
-		authModel := authorizationmodel.AuthzModel{}
-		clientConfig.StoreID = storeID
+	modelWriteRes, err := model.Write(fgaClient, authModel)
+	if err != nil {
+		return nil, fmt.Errorf("failed to write model: %w", err)
+	}
+
+	response.Model = modelWriteRes
+
+	return response, nil
+}
+
+func importStore(
+	clientConfig fga.ClientConfig,
+	fgaClient client.SdkClient,
+	storeData *storetest.StoreData,
+	format authorizationmodel.ModelFormat,
+	storeID string,
+	maxTuplesPerWrite int,
+	maxParallelRequests int,
+	fileName string,
+) (*CreateStoreAndModelResponse, error) {
+	var (
+		response *CreateStoreAndModelResponse
+		err      error
+	)
 
-		err = authModel.ReadModelFromString(storeData.Model, format)
+	if storeID == "" {
+		response, err = createStore(clientConfig, storeData, format, fileName)
 		if err != nil {
-			return nil, err //nolint:wrapcheck
+			return nil, fmt.Errorf("failed to create store: %w", err)
 		}
-
-		modelWriteRes, err := model.Write(fgaClient, authModel)
+	} else {
+		response, err = updateStore(clientConfig, fgaClient, storeData, format, storeID)
 		if err != nil {
-			return nil, fmt.Errorf("failed to write model due to %w", err)
+			return nil, fmt.Errorf("failed to update store: %w", err)
 		}
-
-		response.Model = modelWriteRes
 	}
 
-	fgaClient, err := clientConfig.GetFgaClient()
+	fgaClient, err = clientConfig.GetFgaClient()
 	if err != nil {
-		return nil, fmt.Errorf("failed to initialize FGA Client due to %w", err)
+		return nil, fmt.Errorf("failed to initialize FGA Client: %w", err)
 	}
 
 	writeRequest := client.ClientWriteRequest{
```

Comment:
Is there much gain from having this instantiated once outside the for loop and then updated per loop rather than declaring per loop?

Reply:
Resolved in 98b8157